### DISCUSSION TITLE
How to load model programmatically?

### DISCUSSION THREAD

**Discussion by velociraptor111** [aws/sagemaker-python-sdk#2360] [Help]
I notice that whenever I run a new batch transform jobs, it will create a new model and saves it. 
I can see all the models from my batch transform jobs in my AWS Sagemaker Dashboard/inference/models

Here is the script that I run 
```python
sagemaker_model = MXNetModel(model_data = 's3://' + sagemaker_session.default_bucket() + '/model/yolo_object_person_detector.tar.gz',
                             role = role, 
                             entry_point = 'entry_point.py',
                             py_version='py3',
                             framework_version='1.4.1',
                            sagemaker_session = sagemaker_session)

transformer = sagemaker_model.transformer(instance_count=1, instance_type='ml.m4.xlarge', output_path=batch_output)

transformer.transform(data=batch_input, content_type='application/x-image')

transformer.wait()
```

I've looked into the source code for declaration of class MXNetModel
```python
class MXNetModel(FrameworkModel):
    """An MXNet SageMaker ``Model`` that can be deployed to a SageMaker ``Endpoint``."""

    __framework_name__ = "mxnet"
    _LOWEST_MMS_VERSION = "1.4"

def __init__(
        self,
        model_data,
        role,
        entry_point,
        image=None,
        py_version="py2",
        framework_version=MXNET_VERSION,
        predictor_cls=MXNetPredictor,
        model_server_workers=None,
        **kwargs
    ):

...
``` 

But I am not seeing anywhere where I can simply load the MXNetModel object using a URL Endpoint to the models in my dashboard.

If I go to console and click one of those models, I can see a button for **Create batch transform job**, so I know internally this is possible. But I can't find anything on the docs to do it programmatically.

Also as a side question:
How many models does the Free tier provide? In the free tier page: https://aws.amazon.com/sagemaker/pricing/ it just says the number of hours, but not necessarily the number of models

**Comment by ChoiByungWook** [comment]
Hello @velociraptor111,

As of now, the Python SDK doesn't have the ability to reference a defined model in the SageMaker platform. I have added an item in the backlog (MLFW-2709), as I believe implementing this ability in conjunction with the existing Model class will take some time.

It is possible to do this using the AWS SDK or boto3 directly, however that won't be as convenient. If you do go this route, you would specify the [model name when generating the endpoint configuration](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_endpoint_config).

> Also as a side question:
How many models does the Free tier provide? In the free tier page: https://aws.amazon.com/sagemaker/pricing/ it just says the number of hours, but not necessarily the number of models

Let me reach out to the corresponding team and get back to you on that.

Thanks!

**Comment by laurenyu** [comment]
Just wanted to circle back on this:
> Also as a side question:
How many models does the Free tier provide? In the free tier page: https://aws.amazon.com/sagemaker/pricing/ it just says the number of hours, but not necessarily the number of models

The free tier limit for model deployment is 125 hours of m4.xlarge or m5.xlarge for real-time inference and batch transform, and this usage can be for a single model or aggregated across multiple models.

**Comment by strawberrypie** [comment]
@ChoiByungWook is there any update on loading a model from SageMaker by its name?
